feat/feature_test.go

```diff
@@ -269,16 +289,12 @@ func (s *S) TestBaseOrientationOf(c *check.C) {
 
 func (s *S) TestOrientationWithin(c *check.C) {
 	for _, t := range orientationTests {
-		c.Check(feat.OrientationWithin(t.f, t.ref), check.Equals, t.ori)
-	}
-
-	// Check we can find things that are all along the feature chain.
-	for ref := feat.Feature(pribA); ref != nil; ref = ref.Location() {
-		c.Check(feat.OrientationWithin(pribA, ref), check.Equals, feat.Forward)
+		c.Check(feat.OrientationWithin(t.f, t.ref), check.Equals, t.oriWithin)
 	}
 
 	// Check that unorientable features return different reference features.
 	c.Check(feat.OrientationWithin(pribA, nil), check.Equals, feat.NotOriented)
+	c.Check(feat.OrientationWithin(orfB, chrom1), check.Equals, feat.NotOriented)
```

Comment:
>And one more thing. What about OrientationWithin(freeOri, nil). On the old version it returns Reverse. In the new version it returns NotOriented.

I think the new version is correct.

>And one more thing which I just remembered drove me to this design and that you might want to think about. Imagine if we had an OrientedFeature interface in addition to the Feature. How would you design the OrientationWithin signature then? I thought of OrientationWIthin(f, ref OrientedFeature) Orientation and this drove me to the restriction that ref cannot be a non orienter. But as you said maybe OrientationWithin(f OrientedFeature, ref Feature) Orientation is more fitting.

You can't make this, unless OrientedFeature.Location() returns an OrientedFeature, which means it can't satisfy Feature.